src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -398,10 +419,24 @@ private void setState(State state) {
     this.state = state;
   }
 
+  public void setIsFirst(boolean isFirst) {
+    this.isFirst = isFirst;
+  }
+
+  public void setFlushPartial(boolean flushPartial) {
```

Comment:
It seems that `flushPartial` is only set to `true` in practice and that the `setFlushPartial` call only exists to support the tests. Why have the `flushPartial` flag at all if it only takes on one value? It looks like the same is also true of `isFirst`.

Normally we'd just test the code as a blackbox, feeding it some input and validating that the expected output was generated. Is there a reason that wouldn't work here? It seems like you just want partial data to be written upon flush, which seems like it should be testable without manually exposing some internal flags.